aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1385,6 +1386,8 @@ impl AptosVM {
             ));
         }
 
+        zkid_validation::validate_zkid_authenticators(transaction, resolver, session, log_context)?;
```

Comment:
Can you change this to return the authenticators so that you avoid a repeated call to `get_zkid_authenticators` below?

Reply:
Just replace its `Result<(),...>` return with a `Result<Vec<(ZkIdPublicKey, ZkIdSignature)>>`

Reply:
this refactor not as simple as you imagined

The first call is to check for feature flags at the top level
call stack: **validate_transaction -> get_zkid_authenticators**

The second call is when doing validation
call stack: **validate_transaction -> validate_signed_transaction -> validate_zkid_authenticators -> get_zkid_authenticators**

To avoid the repeated call we would need to bring the feature flag gate into validate_zkid_authenticators, but it's probably where it belongs given that the passkeys flag gating is there (and does the same thing)